src/backend/services/chat.py

```diff
@@ -78,31 +80,46 @@ def process_chat(
     """
     user_id = get_header_user_id(request)
     deployment_name = request.headers.get("Deployment-Name", "")
-    model_config = {}
+    deployment_config = {}
     # Deployment config is the settings for the model deployment per request
     # It is a string of key value pairs separated by semicolons
     # For example: "azure_key1=value1;azure_key2=value2"
+    # TODO Eugene: Confirm it with Scott - header deployment config has priority over the deployment config in the DB,
+    # but agent deployment config has priority over both if it is set
     if not request.headers.get("Deployment-Config", "") == "":
-        model_config = get_deployment_config(request)
+        deployment_config = get_deployment_config(request)
+    else:
+        deployment = deployment_crud.get_deployment_by_name(session, deployment_name)
+        if deployment is not None and deployment.is_available:
+            deployment_config = deployment.default_deployment_config
 
     if agent_id is not None:
         agent = agent_crud.get_agent_by_id(session, agent_id)
-
-        # TODO: @Scott Validation error still needs to be fixed here
-        # ROD: error count: <built-in method error_count of pydantic_core._pydantic_core.ValidationError object at 0xffff703f08b0>
-
-        try:
-            add_agent_to_request_state(request, agent)
-        except ValidationError as exc:
-            print(f"Validation error count: {exc.error_count()}")
-            for err in exc.errors():
-                print(f"ROD: error: {repr(err)}")
-
+        add_agent_to_request_state(request, agent)
         if agent is None:
             raise HTTPException(
                 status_code=404, detail=f"Agent with ID {agent_id} not found."
             )
+        # TODO Eugene: Confirm it with Scott
+        # Check if the agent is associated with the deployment
+        if agent.deployment and agent.deployment.name != deployment_name:
+            # if agent's default deployment is not the same as the request's deployment
+            # we try to find that deployment in the agent's associated deployments and use it
```

Comment:
i think this is fine. but again, referencing my comment above:

maybe we can just stop sending the deployment altogether in the request.